lib/galaxy/config/__init__.py

```diff
@@ -760,6 +773,7 @@ def parse_config_file_options(self, kwargs):
             markdown_export_css=[self._in_config_dir('markdown_export.css')],
             markdown_export_css_pages=[self._in_config_dir('markdown_export_pages.css')],
             markdown_export_css_invocation_reports=[self._in_config_dir('markdown_export_invocation_reports.css')],
+            file_path=[self._in_data_dir('files'), self._in_data_dir('objects')],
```

Comment:
at this point, the value of `self.file_path` *is* `self._in_data_dir('objects') (because the default from the schema is resolved w.r.t `data_dir` before `parse_config_file_options()` is called. So, maybe something like this:
`file_path=[self._in_data_dir('files'), self.file_path)` ?
That way we also don't throw away the default value `objects`.

Reply:
That would look less readable/transparent in my opinion? Maybe best to think of the objects in the config schema as more of a documentation thing in this case? If you insist though I can make the change - I see what you're saying.

Reply:
I agree on readability, but I do think making use of the default value for all config options makes the system more predictable and easier to understand (i.e., all configs processed in a uniform way), which, I feel, is a good tradeoff. 
If we throw away the default here, that would create a special case; and I think limiting those when possible would be a good thing. (Ideally, after some refactoring, I’d handle such simple cases in the schema: the default would be `objects, files`, both resolving to `data_dir` and then listified).
Maybe a simple comment in the code would be sufficient to address readability?

Reply:
I've made the requested change, I think it is not great but it is not a big thing either way.

Reply:
Thank you!